### PR TITLE
Refactor: split TxOutput validation & serialization

### PR DESCRIPTION
**Pull request by ppershing**

### CODE REVIEW

File: src/parsing.ts [hunk 1]
```diff
@@ -720,4 +722,67 @@ export function parseAddressParams(
...
+}
+
+export function parseTxOutput(
+    output: TxOutput,
+    network: Network,
+): ParsedOutput {
+    const amountStr = parseUint64_str(output.amountStr, MAX_LOVELACE_SUPPLY_STR, TxErrors.OUTPUT_INVALID_AMOUNT)
+    const assets = (output.tokenBundle ?? []).map((ag) => parseAssetGroup(ag))
+
+    const hasAddressHex = "addressHex" in output && output.addressHex != null
```
Comment (refi93) [inline src/parsing.ts:749]: isn't the `"addressHex" in output` check redundant? same for `spendingPath` below

Comment (ppershing) [reply src/parsing.ts:749]: No, this is typescript limitation with TxOutput being typed as union with one variant missing the key

File: src/parsing.ts [hunk 2]
```diff
@@ -720,4 +722,67 @@ export function parseAddressParams(
...
+    assets: AssetGroup[]
+    destination: OutputDestination
+}
+
+export function parseTxOutput(
+    output: TxOutput,
+    network: Network,
+): ParsedOutput {
+    const amountStr = parseUint64_str(output.amountStr, MAX_LOVELACE_SUPPLY_STR, TxErrors.OUTPUT_INVALID_AMOUNT)
+    const assets = (output.tokenBundle ?? []).map((ag) => parseAssetGroup(ag))
```
Comment (refi93) [inline src/parsing.ts:747]: I'd keep the "tokenBundle" name to stay consistent with the external interface and established terminology (see https://developers.cardano.org/en/development-environments/native-tokens/native-tokens-glossary/). Also, I find it a bit awkward to have something named "assets" only to assign to it "asset groups"

**Review by refi93 — APPROVED** [review]
one minor naming remark, otherwise LGTM

**Pull request merged**